src/main/java/io/aiven/kafka/connect/opensearch/DocumentIDStrategy.java

```diff
@@ -0,0 +1,153 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.opensearch;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import org.opensearch.action.DocWriteRequest;
+import org.opensearch.action.delete.DeleteRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.index.VersionType;
+
+public enum DocumentIDStrategy {
+
+    NONE("none", "No Doc ID is added",
+         record -> null,
+         (request, record) -> Objects.isNull(record.value()) ? null : request
+    ),
+    
+    RECORD_KEY("record.key", "Generated from the record's key",
+        record -> RecordConverter.convertKey(record.keySchema(), record.key()),
+        (request, record) ->  {
+            request.versionType(VersionType.EXTERNAL);
+            request.version(record.kafkaOffset());
+            return request;
+        }
+    ),
+
+    TOPIC_PARTITION_OFFSET("topic.partition.offset", "Generated as record's ``topic+partition+offset``",
+        record ->  record.topic() + "+" + record.kafkaPartition() + "+" + record.kafkaOffset(),
+        (request, record) -> request
+    );
+
+            
+    private final String name;
+
+    private final String description;
+
+    private final Function<SinkRecord, String> docIdGenerator;
+
+    private final BiFunction<DocWriteRequest<?>, SinkRecord, DocWriteRequest<?>> updateRequest;
+
+    private DocumentIDStrategy(final String name, final String description, 
+                               final Function<SinkRecord, String> docIdGenerator,
+                               final BiFunction<DocWriteRequest<?>, SinkRecord, DocWriteRequest<?>> updateRequest) {
+        this.name = name.toLowerCase(Locale.ROOT);
+        this.description = description;
+        this.docIdGenerator = docIdGenerator;
+        this.updateRequest = updateRequest;
+    }
+
+    @Override
+    public String toString() {
+        return name;
+    }
+
+    public static DocumentIDStrategy fromString(final String name) {
+        for (final DocumentIDStrategy strategy : DocumentIDStrategy.values()) {
+            if (strategy.nameEquals(name)) {
+                return strategy;
+            }
+        }
+        throw new IllegalArgumentException("Unknown Document ID Strategy " + name);
+    }
+
+    public static String[] names() {
```

Comment:
Please make it `private static`, or even remove it, this method is used only in one place https://github.com/aiven/opensearch-connector-for-apache-kafka/pull/139/files#diff-f57f55041b60deaa26a11d81710270e886995c941beaaeaa604aa7f205b83fc1R135